repo/format/content_format.go

```diff
@@ -12,12 +12,13 @@ import (
 
 // ContentFormat describes the rules for formatting contents in repository.
 type ContentFormat struct {
-	Hash               string `json:"hash,omitempty"`                        // identifier of the hash algorithm used
-	Encryption         string `json:"encryption,omitempty"`                  // identifier of the encryption algorithm used
-	ECC                string `json:"ecc,omitempty"`                         // identifier of the ecc algorithm used
-	ECCOverheadPercent int    `json:"eccOverheadPercent,omitempty"`          // space overhead for ecc
-	HMACSecret         []byte `json:"secret,omitempty" kopia:"sensitive"`    // HMAC secret used to generate encryption keys
-	MasterKey          []byte `json:"masterKey,omitempty" kopia:"sensitive"` // master encryption key (SIV-mode encryption only)
+	Hash                   string `json:"hash,omitempty"`                        // identifier of the hash algorithm used
+	Encryption             string `json:"encryption,omitempty"`                  // identifier of the encryption algorithm used
+	ECC                    string `json:"ecc,omitempty"`                         // identifier of the ecc algorithm used
+	ECCOverheadPercent     int    `json:"eccOverheadPercent,omitempty"`          // space overhead for ecc
+	HMACSecret             []byte `json:"secret,omitempty" kopia:"sensitive"`    // HMAC secret used to generate encryption keys
+	MasterKey              []byte `json:"masterKey,omitempty" kopia:"sensitive"` // master encryption key (SIV-mode encryption only)
+	KeyDerivationAlgorithm string `json:"keyDerivationAlgorithm,omitempty"`      // key derivation algorith used to generate keys
```

Comment:
i don't understand how this works, `keyDerivationAlgorithm` must be known beforehand to generate key to decrypt this JSON, so by definition it can't be stored in this JSON, right?

Reply:
When creating a new repository we pass this in as the NewRepositoryOptions. See `repo/initialize.go`.
Its also stored unencrypted in the status.

Reply:
it should ONLY be stored unencrypted in `KopiaRepositoryJSON` and not in `format.ContentFormat` - the latter is encrypted using key derived using the algorithm itself, so it makes no sense.

Reply:
We should move `KeyDerivationAlgorithm` to be a field in `NewRepositoryOptions` instead of a field in `format.ContentFormat`